internal/servers/worker/status.go

```diff
@@ -99,6 +59,34 @@ func (w *Worker) LastStatusSuccess() *LastStatusInformation {
 	return w.lastStatusSuccess.Load().(*LastStatusInformation)
 }
 
+// WaitForNextSuccessfulStatusUpdate waits for the next successful status. It's
+// used by testing (and in the future, shutdown) in place of a more opaque and
+// possibly unnecessarily long sleep for things like initial controller
+// check-in, etc.
+//
+// The timeout is aligned with the worker's status grace period. A nil error
+// means the status was sent successfully.
+func (w *Worker) WaitForNextSuccessfulStatusUpdate() error {
+	w.logger.Debug("waiting for next status report to controller")
+	waitStatusStart := time.Now()
+	ctx, cancel := context.WithTimeout(w.baseContext, w.conf.StatusGracePeriodDuration)
+	defer cancel()
+	for {
+		if err := ctx.Err(); err != nil {
```

Comment:
Another way to do this which feels _to me_ to be more canonical but very well may not be is to have a `select` statement on `ctx.Done()` and `time.After(time.Second)`. This will work, but may be worth looking into.